portal/app/[locale]/bitcoin-yield/_components/cardInfo.tsx

```diff
@@ -0,0 +1,47 @@
+import { Card } from 'components/card'
+import { P } from 'components/paragraph'
+import Image, { StaticImageData } from 'next/image'
+
+type Props<T> = {
+  data: T | undefined
+  formatValue: (value: T) => string
+  isError: boolean
+  icon: StaticImageData
+  label: string
+}
+export const CardInfo = function <T>({
+  data,
+  formatValue,
+  icon,
+  isError,
+  label,
+}: Props<T>) {
+  const getValue = function () {
+    if (isError) {
+      return '-'
+    }
+    if (data === undefined) {
+      return '...'
+    }
+    return formatValue(data)
+  }
```

Comment:
It seems unnecessary to pass a `formatValue` function to the component. The caller could send no data, so the component renders `...` or the formatted data so the component renders that instead. What is the current benefit of injecting that function?